core/src/test/scala/integration/kafka/network/DynamicNumNetworkThreadsTest.scala

```diff
@@ -0,0 +1,102 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package integration.kafka.network
+
+import kafka.server.{BaseRequestTest, Defaults, KafkaConfig}
+import kafka.utils.TestUtils
+import org.apache.kafka.clients.admin.{Admin, AdminClientConfig}
+import org.apache.kafka.common.network.ListenerName
+import org.junit.jupiter.api.Assertions._
+import org.junit.jupiter.api.{BeforeEach, Test, TestInfo}
+
+import java.util.Properties
+import scala.jdk.CollectionConverters._
+
+class DynamicNumNetworkThreadsTest extends BaseRequestTest {
+
+  override def brokerCount = 1
+
+  val internal = "PLAINTEXT"
+  val external = "EXTERNAL"
+
+  override def brokerPropertyOverrides(properties: Properties): Unit = {
+    properties.put(KafkaConfig.ListenersProp, s"$internal://localhost:0, $external://localhost:0")
+    properties.put(KafkaConfig.ListenerSecurityProtocolMapProp, s"$internal:PLAINTEXT, $external:PLAINTEXT")
+    properties.put(s"listener.name.${internal.toLowerCase}.${KafkaConfig.NumNetworkThreadsProp}", "2")
+  }
+
+  @BeforeEach
+  override def setUp(testInfo: TestInfo): Unit = {
+    super.setUp(testInfo)
+    TestUtils.createTopic(zkClient, "test", brokerCount, brokerCount, servers)
+    assertEquals(2, getNumNetworkThreads(internal))
+    assertEquals(Defaults.NumNetworkThreads, getNumNetworkThreads(external))
+  }
+
+  def getNumNetworkThreads(listener: String): Int = {
+    brokers.head.metrics.metrics().keySet().asScala
+      .filter(_.name() == "request-rate")
+      .count(listener == _.tags().get("listener"))
+  }
+
+  @Test
+  def testDynamicNumNetworkThreads(): Unit = {
```

Comment:
Is it worth having a test to cover changing the control plane listener's threads too?

Reply:
The control place always has a single processor thread.